contracts/MarginBase.sol

```diff
@@ -272,6 +276,24 @@ contract MarginBase is MinimalProxyable, IMarginBase, OpsReady {
             // define market via _marketKey
             IFuturesMarket market = futuresMarket(_newPositions[i].marketKey);
 
+            /// @notice (if it exists) update position here to ensure internal
+            /// accounting is up-to-date moving forward
+            /// @dev if position has been liquidated, this contract will
+            /// effectively create a new position as specified by _newPositions[i]
+            if (activeMarketPositions[_newPositions[i].marketKey].size != 0) {
+                fetchPositionAndUpdate(_newPositions[i].marketKey, market);
+            }
+
+            /// @notice if new position in market, ensure size > 0 to prevent
+            /// wasting gas to reach error: "MissingMarketKey"
+            /// @dev this second check is due to above logic potentially changing internal state
+            if (
+                activeMarketPositions[_newPositions[i].marketKey].size == 0 &&
```

Comment:
does this actually work if the position is liquidated? by this point it's already removed from our internal accounting so the size would be 0?

Reply:
Size would be zero (default value after deletion) and we run into the situation (which has always existed) where size for a new position *must* be *non-zero*

Reply:
Before, we just did not check for this. If a new position specified non-zero margin and zero size, then if the position is new (i.e. not modifying a pre-existing one), then it would revert with error: MissingMarketKey().